docs/linux/setup_ubuntu-host_android-arm32.md

```diff
@@ -0,0 +1,123 @@
+# Setup: Ubuntu host, arm32 kernel on an Android device
+
+This document will detail the steps involved in setting up a Syzkaller instance fuzzing an ARM32 linux kernel on an Android (or Android Things) device. This is a work-in-progress at this time and being provided to spur further development. Some features of syzkaller may not yet work on ARM32. For example, not all debugging and test coverage features are available in the Linux kernel for ARM32, limiting the efficacy of syskaller in finding bugs fast. These instructions help set up syzkaller to be a basic fuzzer that does not rely on test coverage data from the kernel. 
+
+## Install Android and Linux kernel on an ARM32 device
+
+Follow the instructions for the ARM32 board to install Android or 
+Android Things and make sure the device boots properly.
+
+Set up the adb bridge so that adb and fastboot work.
+
+Setup a serial port, following the instructions for your board so that you can monitor any messages from the kernel.
+
+These were tested on an NXP Pico-Pi-IMX7D following the instructions at:
+
+```
+
+https://developer.android.com/things/hardware/developer-kits.html.
+
+```
+
+
+If feasible, recompile and reinstall the Linux kernel with any debugging options available on your board.
+
+
+## Optional: Setup the NDK cross-compiler toolchain
+
+
+You can find the NDK packages here:
+
+https://developer.android.com/ndk/downloads/index.html
+
+
+Download the package for your host system (linux-x86_64). After you have unzipped it, you should have a directory named android-ndk-r15c, or the latest stable release. We will refer to this directory as $(NDK).
+
+
+Generate a standalone toolchain for the target device in a destination of your choice (assumed to be defined in the variable NDKARM). The destination folder $NDKARM will get created by the code below.
```

Comment:
Is this step necessary? We don't do this for other archs. Instead we use the toolchain directly and add necessary compiler flags. Let's do the same here.

Reply:
The problem is that clang is not there in the prebuilt toolchain. And using g++ or c++ gives the following error when compiling executor_linux.cc:

/tmp/ccY1c1OE.s: Assembler messages:
/tmp/ccY1c1OE.s:4848: Error: alignment too large: 15 assumed

I could likely  find a way to build the standalone toolchain as part of standard make and then use it to compile executor_linux.cc. Would that be preferable?

Reply:
Using latest NDK:
```
$ wget https://dl.google.com/android/repository/android-ndk-r16-linux-x86_64.zip
$unzip android-ndk-r16-linux-x86_64.zip
$ find android-ndk-r16 -name clang
android-ndk-r16/toolchains/llvm/prebuilt/linux-x86_64/bin/clang
```